lib/shared-class.js

```diff
@@ -212,12 +215,37 @@ SharedClass.prototype.find = function(fn, isStatic) {
  */
 
 SharedClass.prototype.disableMethod = function(fn, isStatic) {
+  deprecated('SharedClass.prototype.disableMethod is deprecated. ' +
+    'Use SharedClass.prototype.disableMethodByName instead.');
   var disableMethods = this._disabledMethods;
   var key = this.getKeyFromMethodNameAndTarget(fn, isStatic);
   disableMethods[key] = true;
 };
 
 /**
+ * Find a sharedMethod with the given static or prototype method name.
+ *
+ * @param {String} methodName The method name
+ * Find a static or prototype method with the given name.
+ * @returns {SharedMethod}
+ */
+
+SharedClass.prototype.findMethodByName = function(methodName) {
+  var isStatic = methodName.indexOf('prototype.') < 0;
+  return this.find(methodName, isStatic);
+};
+
+/**
+ * Disable a sharedMethod with the given static or prototype method name.
+ *
+ * @param {String} methodName The method name
+ */
+
+SharedClass.prototype.disableMethodByName = function(methodName) {
+  this.disableMethod(methodName);
```

Comment:
If this rather hacky. IIUC, it's relying on the fact that `getKeyFromMethodNameAndTarget` will pass through `prototype.name`.